### DISCUSSION TITLE
Caddyfile

### DISCUSSION THREAD

**Discussion by ottobaer** [Zavy86/WikiDocs#145] [Q&A]
I'm currently trying to get it work with Caddy, but I keep on failing with the rewrite rules needed. 

The php part works, but I can't create new pages, it will just edit the homepage. 

This is what I have so far. 

```
https://notes.lan {

    root * /opt/wikidocs
    php_fastcgi unix//run/php-fpm/php-fpm.sock
    file_server


@doc {
	path_regexp ^/(.)$
	not file {path} {path}/
}
rewrite @doc /index.php?doc={1}

}

```

**Comment by leomoon** [comment]
I don't have caddy to test it but how about something like this:
```
https://notes.lan {
    # Static file handling with caching
    @static {
        file {
            try_files {path}
        }
        path *.css *.js *.jpg *.jpeg *.png *.gif *.ico *.svg
    }
    handle @static {
        header Cache-Control "public, max-age=3600"
        file_server
    }

    # PHP handling
    try_files {path} /index.php?doc={path}
    php_fastcgi unix//run/php/php-fpm.sock

    # Fallback for other requests
    file_server
}
```

**Comment by ottobaer** [comment]
Thanks for the help. 

This doesn't seem to work, I still can only edit the homepage. 

![image](https://github.com/user-attachments/assets/554d6b22-0921-47f6-a6d4-6c8fe2d40526)


![image](https://github.com/user-attachments/assets/193bc00f-1f9b-4f6a-ad5c-4f386f8b34a4)

it looks ok, but that only opens the homepage for editing without creating a new page.

**Comment by ottobaer** [comment]
After some fiddling it works, but I don't know if it's correct. 


```
 @rewrites not file
  route @rewrites {
    @pc path_regexp ^/(.*)$
    rewrite @pc /index.php?doc={re.1}&{file}&{query}
   }

```

**Comment by leomoon** [comment]
how about this?
```
https://notes.lan {
  @notFile {
    not file {path}
  }
  rewrite @notFile /index.php?doc={uri}
  file_server
}
```

**Comment by ottobaer** [comment]
just tried that - doesn't work, but I have updated my solution I posted before and it's working :).
